service/notification/NotificationProcessor.php

```diff
@@ -335,4 +354,48 @@ private function isCurrentOrderStatusANonFinalStatus($currentOrderStatus)
 
         return false;
     }
+
+    /**
+     * @param $notification
+     * @param OrderCore $order
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateWithCartAndOrder($notification, OrderCore $order)
+    {
+        $cart = \Cart::getCartByOrderId($order->id);
+
+        if (!\Validate::isLoadedObject($cart)) {
+            $this->logger->addAdyenNotification(
+                sprintf(
+                    'Unable to load cart object linked to Order (%s) and Notification (%s)',
+                    $order->id,
+                    $notification['entity_id']
+                )
+            );
+
+            return false;
+        }
+
+        $cartCurrency = \Currency::getCurrency($cart->id_currency);
+        $orderCurrency = \Currency::getCurrency($order->id_currency);
+        $cartCurrencyIso = $cartCurrency['iso_code'];
+        $orderCurrencyIso = $orderCurrency['iso_code'];
```

Comment:
Are we certain that these arrays will contain those indexes or should we do some defensive programming here with `!empty()`?

Reply:
It's like an object that should always have a value - if not then it's more of a core issue in the shop. Of course we can add some defensive programming, although in that case I'd create an adapter for the object with a function which always does the validation before retrieving the property